acceleration-everest-03-dotnet/Api/Controllers/CustomerController.cs

```diff
@@ -16,71 +17,92 @@ public CustomerController(ICustomerAppService appService)
         _customerAppService = appService ?? throw new ArgumentNullException(nameof(appService));
     }
 
-    [HttpDelete]
-    public IActionResult Delete(long id)
+    [HttpPost]
+    public async Task<IActionResult> CreateAsync(CreateCustomer customerToCreate)
     {
         try
         {
-            _customerAppService.Delete(id);
-            return NoContent();
+            var id = await _customerAppService.CreateAsync(customerToCreate).ConfigureAwait(false); ;
+            
+            return Created("Id:", id);
         }
         catch (ArgumentException exception)
         {
             var message = exception.InnerException?.Message ?? exception.Message;
-            return NotFound(message);
+            
+            return BadRequest(message);
         }
     }
 
-    [HttpGet]
-    public IActionResult GetAll()
+    [HttpDelete]
+    public async Task<IActionResult> DeleteAsync(long customerId)
     {
-        var result = _customerAppService.GetAll();
+        try
+        {
+            await _customerAppService.DeleteAsync(customerId).ConfigureAwait(false);
 
-        return Ok(result);
+            return NoContent();
+        }
+        catch (ArgumentException exception)
+        {
+            var message = exception.InnerException?.Message ?? exception.Message;
+            
+            return NotFound(message);
+        }
     }
 
-    [HttpGet("{id}")]
-    public IActionResult GetById(long id)
+    [HttpGet]
+    public async Task<IActionResult> GetAllCustomersAsync()
     {
         try
         {
-            var result = _customerAppService.GetById(id);
+            var result = await _customerAppService.GetAllCustomersAsync().ConfigureAwait(false);
+
             return Ok(result);
         }
-        catch (ArgumentException exception)
+        catch
         {
-            var message = exception.InnerException?.Message ?? exception.Message;
-            return NotFound(message);
+            return NoContent();
         }
     }
 
-    [HttpPost]
-    public IActionResult Create(CreateCustomerDto customer)
+    [HttpGet("{customerId}")]
+    public async Task<IActionResult> GetByCustomerIdAsync(long customerId)
```

Comment:
Estamos na controller de Customer, não há razão para colocar customer no url, apenas Id
```suggestion
    [HttpGet("{id}")]
    public async Task<IActionResult> GetByCustomerIdAsync(long Id)
```

Reply:
Alterado em todas as controllers